### PR TITLE
Added algo Independent Component Analysis(ICA)

### LINKED ISSUE #1227: [Feature]: Adding new ML Algo Independent Component Analysis (ICA)
### Feature Name

Independent Component Analysis (ICA) in Machine Learning

### Feature Description

- It is an unsupervised learning algorithm that aims to decompose a multivariate signal into independent components.
- This will involve implementing in python using libraries like NumPy and Scikit-learn to demonstrate its application in signal processing.

### Motivation

ICA is widely used in applications like blind source separation, image processing, and EEG/MEG data analysis. By implementing ICA, users will benefit from a method that allows them to isolate underlying independent factors from mixed data, enhancing their ability to analyze complex datasets and uncover hidden patterns.

### Implementation Suggestions (Optional)

_No response_

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### PR DESCRIPTION
**Pull request by siri-chandana-macha**
## 📥 Pull Request

Fixes #1227 

### Description
Added Independent Component Analysis (ICA) with an overview, key concepts, complexity, and Python code example, enhancing the repository’s machine learning resources. Detailed ICA’s role in signal separation applications.

@ajay-dhangar please add all the labels in #1227

### CODE REVIEW

File: docs/machine-learning/Independent Component Analysis.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+4. **Independent Component Extraction**:  
+   ICA algorithms, such as FastICA or Infomax, are used to extract the independent components from the mixed signals. This is typically done by maximizing the statistical independence of the output signals.
+
+### Mathematical Model:
+In ICA, the observed data \( X \) can be modeled as:
+
+\[
+X = A.S
+\]
+
```
Comment (ajay-dhangar) [inline docs/machine-learning/Independent Component Analysis.md]: replace

```md
\[
X = A.S
\]
```
with

```md
$$
X = A.S
$$
```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please read our [contribution guideline](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md ) for a better understanding of our format and expectations. When writing content, please follow the guidelines to ensure that your contributions are accepted. 

And also, can't modify `package-lock.json` file

File: docs/machine-learning/Independent Component Analysis.md (hunk 1 above)
Comment (siri-chandana-macha) [reply docs/machine-learning/Independent Component Analysis.md]: @ajay-dhangar  I’ve updated the math notation  as requested and have reviewed the contribution guidelines.
 Additionally, I’ve ensured no modifications were made to the package-lock.json file.

Please let me know if any further adjustments are needed!

**Comment by siri-chandana-macha** [comment]
@ajay-dhangar I’ve updated the math notation as requested and have reviewed the contribution guidelines.
Additionally, I’ve ensured no modifications were made to the package-lock.json file.

Please let me know if any further adjustments are needed!

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by siri-chandana-macha** [comment]
@ajay-dhangar Can you please add hacktoberfest-accepted label?

**Pull request merged**